src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetCoreMvc2Integration.cs

```diff
@@ -138,55 +55,48 @@ public AspNetCoreMvc2Integration(object actionDescriptor, object httpContext)
             int opCode,
             int mdToken)
         {
-            AspNetCoreMvc2Integration integration = null;
-
             if (!Tracer.Instance.Settings.IsIntegrationEnabled(IntegrationName))
             {
                 // integration disabled
                 return;
             }
 
-            try
-            {
-                integration = new AspNetCoreMvc2Integration(actionDescriptor, httpContext);
+            string methodDef = $"{DiagnosticSource}.{nameof(BeforeAction)}(...)";
+            var integration = WebServerIntegrationContext.RetrieveFromHttpContext(httpContext);
```

Comment:
I think the naming could be slightly more clear if you changed the name of the `integration` variable here and in the other wrapper methods to `integrationContext`.